### DISCUSSION TITLE
a couple qustions, one about xformers , and one for black output.

### DISCUSSION THREAD

**Discussion by whitevamp** [vladmandic/sdnext#1943] [Q&A]
first question:
I just found this repo/ app, and decided to try it out, and it installed everything like it should, and then I noticed this error " No module 'xformers'. Proceeding without it. " as its starting up, before ui startup.
 so question is can this be safely  ignored or ?

second question:
when generating a txt2img I am getting a black output.
I know on stable-diffusion-webui I had this issue after updating to 1.5 and it was because of xformers, and I installed it but it was 
" 23:48:31-290716 WARNING  Not used, uninstalling: xformers 0.0.21.dev578
23:48:31-292716 DEBUG    Running pip: uninstall xformers --yes --quiet "
so what might be this issue with this ?

 here is my log with debug
`
Using VENV: G:\AI_Generation\automatic-webui\venv
23:11:35-246270 INFO     Starting SD.Next
23:11:35-250269 INFO     Python 3.10.6 on Windows
23:11:35-307274 INFO     Version: 24698912 Wed Aug 2 08:27:16 2023 +0200
23:11:39-922870 INFO     Latest published version: fc3d96f6f6a0007edc5f1531f1244cf5034bf9da 2023-08-04T01:15:25Z
23:11:39-923870 DEBUG    Setting environment tuning
23:11:39-924870 DEBUG    Torch overrides: cuda=True rocm=False ipex=False diml=False
23:11:39-925870 DEBUG    Torch allowed: cuda=True rocm=False ipex=False diml=False
23:11:39-929872 INFO     nVidia CUDA toolkit detected
23:11:39-930871 WARNING  Not used, uninstalling: xformers 0.0.21.dev578
23:11:39-931871 DEBUG    Running pip: uninstall xformers --yes --quiet
23:11:41-209644 DEBUG    Repository update time: Tue Aug  1 23:27:16 2023
23:11:41-211149 DEBUG    Previous setup time: Thu Aug  3 21:22:55 2023
23:11:41-213156 INFO     Disabled extensions: []
23:11:41-214154 INFO     Enabled extensions-builtin: ['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'ScuNET', 'sd-dynamic-thresholding',
                         'sd-extension-system-info', 'sd-webui-agent-scheduler', 'sd-webui-controlnet',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR']
23:11:41-226154 INFO     Enabled extensions: []
23:11:41-227153 DEBUG    Latest extensions time: Thu Aug  3 21:22:39 2023
23:11:41-229155 DEBUG    Timestamps: version:1690957636 setup:1691122975 extension:1691122959
23:11:41-230154 INFO     No changes detected: Quick launch active
23:11:41-231154 INFO     Verifying requirements
23:11:41-254154 INFO     Disabled extensions: []
23:11:41-256154 INFO     Enabled extensions-builtin: ['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'ScuNET', 'sd-dynamic-thresholding',
                         'sd-extension-system-info', 'sd-webui-agent-scheduler', 'sd-webui-controlnet',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR']
23:11:41-270154 INFO     Enabled extensions: []
23:11:41-277154 INFO     Extension preload: 0.0s G:\AI_Generation\automatic-webui\extensions-builtin
23:11:41-278154 INFO     Extension preload: 0.0s G:\AI_Generation\automatic-webui\extensions
23:11:41-300159 DEBUG    Memory used: 0.04 total: 31.92 Collected 0
23:11:41-301158 DEBUG    Starting module: <module 'webui' from 'G:\\AI_Generation\\automatic-webui\\webui.py'>
23:11:41-303158 INFO     Server arguments: ['--lowvram', '--autolaunch', '--use-cuda', '--debug']
23:11:41-330668 DEBUG    Loading Torch
23:11:48-725564 DEBUG    Loading Gradio
23:11:49-752717 DEBUG    Loading Modules
No module 'xformers'. Proceeding without it.
23:11:51-379588 DEBUG    Reading: G:\AI_Generation\automatic-webui\config.json len=296
23:11:51-382588 INFO     Pipeline: Backend.ORIGINAL
23:11:51-384588 DEBUG    Loaded styles: G:\AI_Generation\automatic-webui\styles.csv 0
23:11:51-587259 DEBUG    Samplers enumerated: ['UniPC', 'DDIM', 'PLMS', 'Euler a', 'Euler', 'DPM++ 2S a', 'DPM++ 2S a
                         Karras', 'DPM++ 2M', 'DPM++ 2M Karras', 'DPM++ SDE', 'DPM++ SDE Karras', 'DPM++ 2M SDE', 'DPM++
                         2M SDE Karras', 'DPM fast', 'DPM adaptive', 'DPM2', 'DPM2 Karras', 'DPM2 a', 'DPM2 a Karras',
                         'LMS', 'LMS Karras', 'Heun']
23:11:51-642775 INFO     Libraries loaded
23:11:51-644775 DEBUG    Entering start sequence
23:11:51-799705 DEBUG    Version: {'app': 'sd.next', 'updated': '2023-08-02', 'hash': '24698912', 'url':
                         'https://github.com/vladmandic/automatic.git/tree/master'}
23:11:51-801705 INFO     Using data path: G:\AI_Generation\automatic-webui
23:11:51-803706 DEBUG    Event loop: <_WindowsSelectorEventLoop running=False closed=False debug=False>
23:11:51-804705 DEBUG    Entering initialize
23:11:51-844216 INFO     Available VAEs: G:\AI_Generation\automatic-webui\models\VAE 6
23:11:51-870216 DEBUG    Reading: G:\AI_Generation\automatic-webui\cache.json len=2
23:11:52-608433 DEBUG    Reading: G:\AI_Generation\automatic-webui\metadata.json len=2606
23:11:52-725462 INFO     Available models: G:\AI_Generation\automatic-webui\models\Stable-diffusion 146
23:11:52-827973 DEBUG    Loading scripts
23:11:56-914694 INFO     ControlNet v1.1.234
ControlNet v1.1.234
ControlNet preprocessor location: G:\AI_Generation\automatic-webui\extensions-builtin\sd-webui-controlnet\annotator\downloads
23:11:57-078717 INFO     ControlNet v1.1.234
ControlNet v1.1.234
23:11:58-102650 DEBUG    Scripts load: ['a1111-sd-webui-lycoris:2.686s', 'clip-interrogator-ext:0.053s', 'Lora:0.53s',
                         'sd-extension-system-info:0.079s', 'sd-webui-agent-scheduler:0.401s',
                         'sd-webui-controlnet:0.397s', 'stable-diffusion-webui-images-browser:0.124s',
                         'stable-diffusion-webui-rembg:0.808s']
Scripts load: ['a1111-sd-webui-lycoris:2.686s', 'clip-interrogator-ext:0.053s', 'Lora:0.53s', 'sd-extension-system-info:0.079s', 'sd-webui-agent-scheduler:0.401s', 'sd-webui-controlnet:0.397s', 'stable-diffusion-webui-images-browser:0.124s', 'stable-diffusion-webui-rembg:0.808s']
23:12:02-200812 INFO     No internet access detected, using default fonts
23:12:02-203597 INFO     Loading UI theme: name=amethyst-nightfall style=Auto
23:12:02-207602 DEBUG    Creating UI
23:12:02-264112 DEBUG    Reading: G:\AI_Generation\automatic-webui\ui-config.json len=0
23:12:02-498140 DEBUG    Extra networks: checkpoints items=146 subdirs=42
23:12:02-572251 INFO     Extra network created thumbnails: checkpoints 0
23:12:10-959044 DEBUG    Extra networks: lora items=2415 subdirs=136
23:12:11-277584 DEBUG    Extra networks: lycoris items=59 subdirs=56
23:12:11-284583 DEBUG    Extra networks: hypernetworks items=1 subdirs=2
23:12:11-330100 DEBUG    UI interface: tab=txt2img batch=False seed=False advanced=False second_pass=False
23:12:11-331101 INFO     Extra network created thumbnails: hypernetworks 0
23:12:11-400100 INFO     Extra network created thumbnails: lycoris 0
23:12:11-540128 INFO     Extra network created thumbnails: lora 0
23:12:11-675641 DEBUG    UI interface: tab=img2img seed=False resize=False batch=False denoise=True advanced=False
23:12:12-065378 DEBUG    Reading: G:\AI_Generation\automatic-webui\ui-config.json len=0
23:12:12-721606 DEBUG    Script: 0.46s ui_tabs
                         G:\AI_Generation\automatic-webui\extensions-builtin\stable-diffusion-webui-images-browser\scrip
                         ts\image_browser.py
23:12:12-725113 DEBUG    Extensions list loaded: G:\AI_Generation\automatic-webui\html\extensions.json
23:12:13-752748 DEBUG    Extension list refresh: processed=203 installed=13 enabled=13 disabled=0 visible=203 hidden=0
Running on local URL:  http://127.0.0.1:7860
23:12:14-313937 INFO     Local URL: http://127.0.0.1:7860/
23:12:14-317938 DEBUG    Gradio registered functions: 1854
23:12:14-319937 INFO     Initializing middleware
23:12:14-323939 DEBUG    Creating API
23:12:14-464973 INFO     [AgentScheduler] Task queue is empty
23:12:14-466975 INFO     [AgentScheduler] Registering APIs
23:12:14-973212 DEBUG    Script: 0.44s app_started
                         G:\AI_Generation\automatic-webui\extensions-builtin\sd-webui-agent-scheduler\scripts\task_sched
                         uler.py
23:12:14-975212 DEBUG    Scripts setup: ['Tiled Diffusion:0.023s', 'ControlNet:0.038s', 'Alternative:0.011s']
23:12:14-977212 DEBUG    Scripts components: []
23:12:14-978212 DEBUG    Model metadata: G:\AI_Generation\automatic-webui\metadata.json no changes
23:12:14-980211 DEBUG    Select checkpoint: model 3DMM_V12.safetensors [8a419d0aa6]
23:12:14-982212 DEBUG    Load model weights: existing=False
                         target=G:\AI_Generation\automatic-webui\models\Stable-diffusion\3DMM_V12.safetensors info=None
23:12:15-371270 DEBUG    gc: collected=2790 device=cuda {'ram': {'used': 1.33, 'total': 31.92}, 'gpu': {'used': 1.05,
                         'total': 8.0}, 'retries': 0, 'oom': 0}
Loading weights: G:\AI_Generation\automatic-webui\models\Stable-diffusion\3DMM_V12.safetensors ━━━━━━━━ 0.0/37.9 -:--:--
                                                                                                        MB
23:12:15-469783 DEBUG    Load model: name=G:\AI_Generation\automatic-webui\models\Stable-diffusion\3DMM_V12.safetensors
                         dict=True
23:12:15-471783 DEBUG    Verifying Torch settings
23:12:15-501784 DEBUG    Torch FP16 test passed
23:12:15-502784 INFO     Torch override dtype: no-half set
23:12:15-504785 INFO     Torch override VAE dtype: no-half set
23:12:15-505786 DEBUG    Desired Torch parameters: dtype=FP16 no-half=True no-half-vae=True upscast=False
23:12:15-506784 INFO     Setting Torch parameters: dtype=torch.float32 vae=torch.float32 unet=torch.float32
23:12:15-507784 DEBUG    Torch default device: cuda
23:12:15-509789 DEBUG    Model dict loaded: {'ram': {'used': 1.35, 'total': 31.92}, 'gpu': {'used': 1.08, 'total': 8.0},
                         'retries': 0, 'oom': 0}
23:12:15-522790 DEBUG    Model config loaded: {'ram': {'used': 1.35, 'total': 31.92}, 'gpu': {'used': 1.08, 'total':
                         8.0}, 'retries': 0, 'oom': 0}
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
23:12:22-010242 DEBUG    Model created from config: G:\AI_Generation\automatic-webui\configs\v1-inference.yaml
23:12:22-013247 DEBUG    Model weights loading: {'ram': {'used': 2.63, 'total': 31.92}, 'gpu': {'used': 1.08, 'total':
                         8.0}, 'retries': 0, 'oom': 0}
23:12:22-035769 DEBUG    Model weights loaded: {'ram': {'used': 2.63, 'total': 31.92}, 'gpu': {'used': 1.08, 'total':
                         8.0}, 'retries': 0, 'oom': 0}
23:12:22-039771 DEBUG    Model weights moved: {'ram': {'used': 2.63, 'total': 31.92}, 'gpu': {'used': 1.08, 'total':
                         8.0}, 'retries': 0, 'oom': 0}
23:12:22-049772 INFO     Applying scaled dot product cross attention optimization
23:12:22-058768 INFO     Embeddings: loaded=0 skipped=0
23:12:22-068769 INFO     Model loaded in 6.7s (load=0.1s config=0.1s create=6.5s)
23:12:22-570370 DEBUG    gc: collected=280 device=cuda {'ram': {'used': 2.63, 'total': 31.92}, 'gpu': {'used': 1.08,
                         'total': 8.0}, 'retries': 0, 'oom': 0}
23:12:22-572368 INFO     Model load finished: {'ram': {'used': 2.63, 'total': 31.92}, 'gpu': {'used': 1.08, 'total':
                         8.0}, 'retries': 0, 'oom': 0} cached=0
23:12:23-028216 DEBUG    gc: collected=124 device=cuda {'ram': {'used': 2.63, 'total': 31.92}, 'gpu': {'used': 1.08,
                         'total': 8.0}, 'retries': 0, 'oom': 0}
23:12:23-030218 INFO     Startup time: 41.7s (torch=7.4s gradio=1.0s libraries=1.9s vae=0.2s models=0.9s codeformer=0.1s
                         scripts=5.3s onchange=4.1s ui-txt2img=9.1s ui-img2img=0.3s ui-train=0.3s ui-settings=0.1s
                         ui-extensions=1.6s ui-defaults=0.1s launch=0.5s app-started=0.7s checkpoint=8.1s)
23:12:23-034215 INFO     Launching browser
23:14:00-198233 DEBUG    Server alive=True Requests=47 memory used: 2.62 total: 31.92
23:16:00-384451 DEBUG    Server alive=True Requests=83 memory used: 2.62 total: 31.92
23:17:59-629692 DEBUG    Server alive=True Requests=118 memory used: 2.62 total: 31.92
23:19:59-737751 DEBUG    Server alive=True Requests=154 memory used: 2.62 total: 31.92

`

**Comment by vladmandic** [comment]
If you want to use xformers, enable them in settings and app will install and use them.
If they are not enabled (default is SDP, not xformers), then app will uninstall them because they can interfere with some modules.
